src/controllers/hid/hidiothread.cpp

```diff
@@ -0,0 +1,301 @@
+#include "controllers/hid/hidiothread.h"
+
+#include <hidapi.h>
+
+#include <QTimer>
+
+#include "controllers/defs_controllers.h"
+#include "controllers/hid/legacyhidcontrollermappingfilehandler.h"
+#include "moc_hidiothread.cpp"
+#include "util/string.h"
+#include "util/time.h"
+#include "util/trace.h"
+
+namespace {
+constexpr int kReportIdSize = 1;
+constexpr int kMaxHidErrorMessageSize = 512;
+constexpr int kSleepTimeWhenIdleMicros =
+        250; // Sleep time of run loop, when no time consuming operation was executed
+constexpr int kWaitForThreadStateIntervalMicros = 500;
+QString loggingCategoryPrefix(const QString& deviceName) {
```

Comment:
Adding a blank line would keep the code readable. Prefer to insert blank lines and placing comments above the respective constant instead of creating those overlong lines.